src/angularComponent/ngcOmniboxSuggestionsDirective.js

```diff
@@ -1,14 +1,95 @@
 import NgcOmniboxSuggestionsController from './ngcOmniboxSuggestionsController.js';
 
-export default function ngcOmniboxSuggestionsDirective() {
+ngcOmniboxSuggestionsDirective.$inject = ['$document', '$templateCache'];
+export default function ngcOmniboxSuggestionsDirective($document, $templateCache) {
+
+  /**
+   * Takes in the public API template submitted by the app-maker and modifies it to repeat through
+   * the data structure of the Omnibox. The template will be different depending on whether it finds
+   * an ngc-omnibox-suggestion-category or not.
+   *
+   * Markup with an ngc-omnibox-suggestion-category:
+   *   <div ng-repeat="suggestion in suggestion.children || omnibox.suggestions">
+   *     <dl ngc-omnibox-suggestion-category ng-if="suggestion.children">
+   *       <dt>
+   *         <h5>{{suggestion.sample_category_title}}</h5>
+   *       </dt>
+   *       <dd>
```

Comment:
I don't see any `<dd>` elements being created in the actual function. This HTML looks more like app-maker provided HTML rather than the structure that the function that's being documented is creating. This confuses me. Can we simplify to **just** what the function creates, with some pseudo-code if necessary? Like:

```
<category-container>
  <category-el>
  <item-children-el>
</category-container>
```